.github/workflows/ci.yaml

```diff
@@ -131,6 +131,19 @@ jobs:
       # Check that compiles for the supported linux targets (aarch64)
       - run: cargo-zigbuild clippy --lib --bins --all-features --target aarch64-unknown-linux-gnu -- -Wclippy::indexing_slicing -D warnings
 
+  # echo "branch=$([[ ${{ github.head_ref }} =~ "^[0-9]+\.[0-9]+\.[0-9]+$" ]]  && echo "true" || echo "false" )" >> "$GITHUB_OUTPUT"
+  check_if_release_branch:
+    runs-on: ubuntu-latest
+    outputs:
+      release_branch: ${{ steps.release-branch.outputs.branch }}
+    steps:
+      - id: release-branch
```

Comment:
Can you explain what's happening in this step and the next?

Reply:
in this step, we get the head ref and compare it with a regex to check if we are on a branch of pattern like `*.*.*`

```
[[ ide-workflow-trigger ~= "^[0-9]+\.[0-9]+\.[0-9]+$" ]]  && echo "true" || echo "false"
```
would fail, while something like `3.5.1` would pass